### PR TITLE
[WIP]Add mechanism to calculate dynamic threshold

### PR DESCRIPTION
**Pull request by michalbiesek**
- add internal threshold limit
- calculate periodically the current size

---
This change is [Reviewable](https://reviewable.io/reviews/pmem/redis/148)

### CODE REVIEW

File: src/pmem.c [hunk 1]
```diff
@@ -49,3 +56,24 @@ void pmemThresholdInit(void)
             serverAssert(NULL);
     }
 }
+
+void adjustPmemThresholdCycle(void) {
+    if (server.memory_alloc_policy == MEM_POLICY_RATIO) {
+        run_with_period(100) {
+            //revert logic to avoid division by zero
+            float setting_state = server.dram_pmem_ratio.pmem_val/server.dram_pmem_ratio.dram_val;
+            float curent_state = zmalloc_used_pmem_memory()/zmalloc_used_memory();
```
Comment (jschmieg) [inline src/pmem.c]: current

Comment (jschmieg) [reply src/pmem.c]: float curent_state = (float)zmalloc_used_pmem_memory()/zmalloc_used_memory();
to get float as a result.

**Review by michalbiesek — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/redis/148)* status: 0 of 5 files reviewed, 1 unresolved discussion (waiting on @jschmieg)

___
*[src/pmem.c, line 65 at r1](https://reviewable.io/reviews/pmem/redis/148#-M3IfctN4p4xvmaWrDhl-r1-65:-M3LkGwa64Piyt_9Fabo:b-896fix) ([raw file](https://github.com/pmem/redis/blob/c3db613288301abc0693cec620e300965a75c671/src/pmem.c#L65)):*
<details><summary><i>Previously, jschmieg wrote…</i></summary><blockquote>

float curent_state = (float)zmalloc_used_pmem_memory()/zmalloc_used_memory();
to get float as a result. 
</blockquote></details>

Done.

**Comment by michalbiesek** [comment]
Job is continued in #158

**Pull request closed without merging**